config/src/main/java/org/springframework/security/config/annotation/web/builders/HttpSecurity.java

```diff
@@ -2653,6 +2653,7 @@ public HttpSecurity addFilterBefore(Filter filter, Class<? extends Filter> befor
 	private HttpSecurity addFilterAtOffsetOf(Filter filter, int offset, Class<? extends Filter> registeredFilter) {
 		int order = this.filterOrders.getOrder(registeredFilter) + offset;
 		this.filters.add(new OrderedFilter(filter, order));
+		this.filterOrders.add(filter.getClass());
```

Comment:
This fix should be added to the `[#addFilter(Filter)](https://github.com/spring-projects/spring-security/pull/9832/files#diff-582061af3df753b11cdf7e63b3c2090816c59ef44480bf0c46523354fe3b5fc5L2666)` method just below this one, shouldn't it?

Reply:
Yea, you are aright. I've changed the line like you suggested.

Reply:
Oh wait, I thought that both methods need the fix 😅

Reply:
Sorry, for that misconception 😅 .